sql/parser/walk.go

```diff
@@ -38,14 +38,15 @@ type Visitor interface {
 	VisitPost(expr Expr) (newNode Expr)
 }
 
-func copyQualifiedNames(q QualifiedNames) QualifiedNames {
-	if q == nil {
+// FIXME(knz) Is this really necessary? It seems that unresolved name
```

Comment:
If we never change them, yeah that can be removed (and "change" I mean both the interface references inside the slices, and the objects they refer).

Note that as is this function doesn't truly make a copy - the new `UnresolvedNames` "points" to the same `[]NamePart`.

We should call out the immutability explicitly where we define `UnresolvedNames`. It is a slice of slices of interface pointers so it's pretty much impossible to see or verify the immutability just from looking at the code.